tests/ci/release.py

```diff
@@ -35,32 +66,45 @@ def run(self, cmd: str, cwd: Optional[str] = None) -> str:
         logging.info("Running command%s:\n    %s", cwd_text, cmd)
         return self._git.run(cmd, cwd)
 
+    def set_release_branch(self):
+        # Get the actual version for the commit before check
+        with self._checkout(self.release_commit, True):
+            self.update()
+            self.release_branch = f"{self.version.major}.{self.version.minor}"
+
+        self.update()
+
     def update(self):
         self._git.update()
         self.version = get_version_from_repo()
 
-    def do(self, args: argparse.Namespace):
-        self.release_commit = args.commit
+    def do(self, no_check_dirty: bool, no_check_branch: bool, no_prestable: bool):
 
-        if not args.no_check_dirty:
+        if not no_check_dirty:
             logging.info("Checking if repo is clean")
             self.run("git diff HEAD --exit-code")
 
-        if not args.no_check_branch:
-            self.check_branch(args.release_type)
+        self.set_release_branch()
+
+        if not no_check_branch:
+            self.check_branch()
 
-        if args.release_type in self.BIG:
-            # Checkout to the commit, it will provide the correct current version
-            with self._checkout(self.release_commit, True):
-                if args.no_prestable:
+        with self._checkout(self.release_commit, True):
+            if self.release_type in self.BIG:
+                # Checkout to the commit, it will provide the correct current version
+                if no_prestable:
                     logging.info("Skipping prestable stage")
                 else:
-                    with self.prestable(args):
+                    with self.prestable():
                         logging.info("Prestable part of the releasing is done")
 
-                with self.testing(args):
+                with self.testing():
```

Comment:
Looks like we can have both testing and prestable release for a single script run? Testing release shouldn't be `BIG` or `SMALL`...

Reply:
Discussed -- add a comment.

Reply:
it's all ready for a single script running right after the prestable part.